compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -124,6 +124,28 @@ static bool isSmallerThan(ArrayRef<int64_t> sourceShape,
                       });
 }
 
+/// Returns the size and offset scaled by some scale factor, and clamped to a
+/// dimSize for the dimension. `(offset + size) * scale` will be clamped to the
+/// `dimSize`.
+static std::pair<OpFoldResult, OpFoldResult>
+getScaledSizeAndOffset(OpBuilder &builder, Location loc, OpFoldResult size,
+                       OpFoldResult offset, int64_t dimSize, int64_t scale) {
+  AffineExpr dim0, dim1;
+  auto ctx = builder.getContext();
+  bindDims(ctx, dim0, dim1);
+  AffineMap offsetMap = AffineMap::get(1, 0, {dim0 * scale}, ctx);
+  Value imageOffset = builder.createOrFold<affine::AffineApplyOp>(
+      loc, offsetMap, getValueOrCreateConstantIndexOp(builder, loc, offset));
```

Comment:
nit: can we use makeComposedFoldedAffineApply here? It would simplify the IR a lot given that `dimSize` and `scale` (and potentially `size` and `offset`) are constants.